internal/provider/vdc/vdc_group_datasource.go

```diff
@@ -0,0 +1,275 @@
+	_ datasource.DataSourceWithConfigure = &vdcGroupDataSource{}
+)
+
+func NewVDCGroupDataSource() datasource.DataSource {
+	return &vdcGroupDataSource{}
+}
+
+type vdcGroupDataSource struct {
+	client *client.CloudAvenue
+}
+
+type vdcGroupDataSourceModel struct {
+	ID                         types.String `tfsdk:"id"`
+	Name                       types.String `tfsdk:"name"`
+	Description                types.String `tfsdk:"description"`
+	ErrorMessage               types.String `tfsdk:"error_message"`
+	DfwEnabled                 types.Bool   `tfsdk:"dfw_enabled"`
+	LocalEgress                types.Bool   `tfsdk:"local_egress"`
+	NetworkPoolID              types.String `tfsdk:"network_pool_id"`
+	NetworkPoolUniversalID     types.String `tfsdk:"network_pool_universal_id"`
+	NetworkProviderType        types.String `tfsdk:"network_provider_type"`
+	Status                     types.String `tfsdk:"status"`
+	Type                       types.String `tfsdk:"type"`
+	UniversalNetworkingEnabled types.Bool   `tfsdk:"universal_networking_enabled"`
+	Vdcs                       types.List   `tfsdk:"vdcs"`
+}
+
+type vdcModel struct {
+	FaultDomainTag       types.String `tfsdk:"fault_domain_tag"`
+	NetworkProviderScope types.String `tfsdk:"network_provider_scope"`
+	IsRemoteOrg          types.Bool   `tfsdk:"is_remote_org"`
+	Status               types.String `tfsdk:"status"`
+	SiteID               types.String `tfsdk:"site_name"`
+	SiteName             types.String `tfsdk:"site_id"`
+	Name                 types.String `tfsdk:"name"`
+	ID                   types.String `tfsdk:"id"`
+}
+
+var vdcModelAttrTypes = map[string]attr.Type{
+	"fault_domain_tag":       types.StringType,
+	"network_provider_scope": types.StringType,
+	"is_remote_org":          types.BoolType,
+	"status":                 types.StringType,
+	"site_id":                types.StringType,
+	"site_name":              types.StringType,
+	"name":                   types.StringType,
+	"id":                     types.StringType,
+}
+
+func (d *vdcGroupDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_" + "group"
+}
+
+func (d *vdcGroupDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides a data source to read vDC group and reference in other resources.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("name")),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of the vDC Group.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("id")),
+				},
+			},
```

Comment:
In the terraform documentation generator the `ID` attribute is only `Computed`. 
Suddenly in the documentation the attribute is in `Read-Only` section. 

I suggest to create new attribute `vdc_group_id`

```suggestion
			"id": schema.StringAttribute{
				// Add MarkdownDescription
				Required: true,
			},
			"vdc_group_id": schema.StringAttribute{
				Optional: true,
				Computed: true,
				Validators: []validator.String{
					stringvalidator.ExactlyOneOf(path.MatchRoot("vdc_group_name")),
				},
			},
			"vdc_group_name": schema.StringAttribute{
				MarkdownDescription: "The name of the vDC Group.",
				Optional:                true,
				Computed:            true,
				Validators: []validator.String{
					stringvalidator.ExactlyOneOf(path.MatchRoot("vdc_group_id")),
				},
			},
```